### DISCUSSION TITLE
Issue importing QTableProps

### DISCUSSION THREAD

**Discussion by eduardoconviso** [quasarframework/quasar#16343] [General - Components / Directives / etc] [answered]
I am creating a QTable wrapper. With that in mind, I have a pretty simple component so far:

```html
<script lang="ts" setup>
import { QTable, type QTableProps } from 'quasar'

withDefaults(defineProps< QTableProps >(), {})
</script>

<template>
  <q-table> </q-table>
</template>

<style lang="scss" scoped></style>
```

Just by doing I have the following error:

![image](https://github.com/quasarframework/quasar/assets/135843115/ce683779-dd03-49cb-b361-f4b9472c3f8f)

I am using Vite 4.4.9, @quasar/vite-plugin 1.4.1

Any ideas on how to fix it?

**Comment by metalsadman** [comment]
you need vue 3.3.x `defineProps<QTableProps>()`

**Reply by eduardoconviso** [reply]
Sorry, got wrong version of the code. Updating with the interface usage. I am using Vue 3.3.4, btw. I still get the same error

**Reply by eduardoconviso** [reply]
It looks like the editor is hiding the `QTableProps` from my text when I use the < and >. Let me upload a picture as well

**Reply by eduardoconviso** [reply]
any ideas with the update @metalsadman ?

**Comment by yusufkandemir** [comment, marked as answer]
This is a Vue problem. This is not working as well:
```html
<script lang="ts" setup>
defineProps<{
  "onUpdate:x"?: (x: number) => void;
  "onUpdate:y"?: (y: string) => void;
}>()
</script>

<template>
  <div />
</template>
```
```
 Expected "}" but found ":"
  5  |    props: {
  6  |      "onUpdate:x": { type: Function, required: false },
  7  |      onUpdate:y: { type: Function, required: false }
     |                ^
  8  |    },
  9  |    setup(__props: any, { expose: __expose }) {
```
I haven't searched the Vue repo to see if it's reported or not. I am not sure which specific Vue version broke it, I am pretty sure this used to work.

**Reply by danielbender1989** [reply]
There is an issue in the Vue repo: https://github.com/vuejs/core/issues/8910
And there might be an upcoming fix: https://github.com/vuejs/core/pull/8654

**Comment by danielbender1989** [comment]
Not a sustainable fix but a way to work around it would be to Omit the problematic attributes:

```
<script lang="ts" setup>
import { QTable, type QTableProps } from 'quasar';

withDefaults(
  defineProps<
    Omit<
      QTableProps,
      'onUpdate:pagination' | 'onUpdate:selected' | 'onUpdate:expanded'
    >
  >(),
  {}
);
</script>

<template>
  <q-table> </q-table>
</template>

<style lang="scss" scoped></style>

```
